### DISCUSSION TITLE
Code Freezing?

### DISCUSSION THREAD

**Discussion by mmliam** [micropython/micropython#9744] [General]
My application uses an ESP8266 that is normally in deep-sleep, wakes to an event, executes several 'imports', processes the event related code and goes back into deep-sleep. The processing of the event related code appears delayed, and the imports are suspect.

Given that the imports are executed at every boot-up, would I be correct in assuming that the overall process would be faster if the code were frozen?

**Comment by robert-hh** [comment]
Frozen code imports much faster (almost instantly) and does not use RAM for the code or for compilation.

**Reply by mmliam** [reply]
Thanks robert,

I've never frozen code before. I have a couple of descriptive links; but I may be back if I have issues - lol.

**Reply by jimmo** [reply]
We recently updated the docs for this -- https://docs.micropython.org/en/latest/reference/manifest.html

Please let us know (or send PRs!) if theres anything that can be improved further.
